packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -337,6 +337,25 @@ Future<vm_service.VmService> setUpVmService({
     registrationRequests.add(
       vmService.registerService(kFlutterGetAndroidBuildVariantsServiceName, kFlutterToolAlias),
     );
+
+    vmService.registerServiceCallback('flutterGetIOSDeeplinkSettings', (Map<String, Object?> params) async {
+      final XcodeUniversalLinkSettings settings = await flutterProject.ios.universalLinkSettings(
+        configuration: params['configuration']! as String,
+        scheme: params['scheme']! as String,
+        target: params['target']! as String,
```

Comment:
I forget how this tool is designed, how will you know what params to pass into this service?

Reply:
The caller will first need to call `flutterGetIOSBuildOptions` vm services to get all of the possible build options, and then they can pick a combination and call this service

Reply:
Can a test be written for this in packages/flutter_tools/test/general.shard/vmservice_test.dart?

Reply:
The flutterGetIOSBuildOptions and this pr are two complete different vm services, and we already have coverage over flutterGetIOSBuildOptions. If such test should exist, I think it should be written in the devtool when it will use these two services. WDYT?

Reply:
Good point, that makes sense. I was just thinking it could be good to have a test connecting the two so that if someone in the future tweaks one, the test will fail and make them aware that `flutterGetIOSDeeplinkSettings` relies on `flutterGetIOSBuildOptions`. But I get your point that technically it doesn't rely on it until it's implemented in the devtool so the testing may be better added then.